frame/plasm-rewards/src/lib.rs

```diff
@@ -0,0 +1,428 @@
+        pub HistoryDepth get(fn history_depth) config(): u32 = 84;
+
+        /// A mapping from still-bonded eras to the first session index of that era.
+        ///
+        /// Must contains information for eras for the range:
+        /// `[active_era - bounding_duration; active_era]`
+        pub BondedEras: Vec<(EraIndex, SessionIndex)>;
+
+        /// The current era index.
+        ///
+        /// This is the latest planned era, depending on how session module queues the validator
+        /// set, it might be active or not.
+        pub CurrentEra get(fn current_era): Option<EraIndex>;
+
+        /// The active era information, it holds index and start.
+        ///
+        /// The active era is the era currently rewarded.
+        /// Validator set of this era must be equal to `SessionInterface::validators`.
+        pub ActiveEra get(fn active_era): Option<ActiveEraInfo<MomentOf<T>>>;
+
+        /// The session index at which the era start for the last `HISTORY_DEPTH` eras
+        pub ErasStartSessionIndex get(fn eras_start_session_index):
+            map hasher(twox_64_concat) EraIndex => Option<SessionIndex>;
+
+        /// True if the next session change will be a new era regardless of index.
+        pub ForceEra get(fn force_era) config(): Forcing;
+
+        /// Storage version of the pallet.
+        ///
+        /// This is set to v2.0.0 for new networks.
+        StorageVersion build(|_: &GenesisConfig| Releases::V2_0_0): Releases;
+    }
+}
+
+decl_event!(
+    pub enum Event<T>
+    where
+        Balance = BalanceOf<T>,
+    {
+        /// The whole reward issued in that Era.
+        /// (era_index: EraIndex, reward: Balance)
+        WholeEraReward(EraIndex, Balance),
+    }
+);
+
+decl_error! {
+    /// Error for the staking module.
+    pub enum Error for Module<T: Trait> {
+        /// Duplicate index.
+        DuplicateIndex,
+        /// Invalid era to reward.
+        InvalidEraToReward,
+    }
+}
+
+decl_module! {
+    pub struct Module<T: Trait> for enum Call where origin: T::Origin {
+        /// Number of sessions per era.
+        const SessionsPerEra: SessionIndex = T::SessionsPerEra::get();
+
+        type Error = Error<T>;
+
+        fn deposit_event() = default;
+
+        fn on_runtime_upgrade() {
+            migrate::<T>();
+        }
+
+        /// On finalize is called at after rotate session.
+        fn on_finalize() {
+            // Set the start of the first era.
+            if let Some(mut active_era) = Self::active_era() {
+                if active_era.start.is_none() {
+                    active_era.start = Some(T::Time::now());
+                    <ActiveEra<T>>::put(active_era);
+                }
+            }
+        }
+
+        // ----- Root calls.
+        /// Force there to be no new eras indefinitely.
+        ///
+        /// # <weight>
+        /// - No arguments.
+        /// # </weight>
+        #[weight = SimpleDispatchInfo::FixedNormal(5_000)]
```

Comment:
May be `FixedOperational` is better for root? https://github.com/paritytech/substrate/blob/master/frame/balances/src/lib.rs#L464

Reply:
I agree. Root call is `operational`.